cli/src/main/java/org/aya/cli/repl/RenderOptions.java

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) 2020-2022 Tesla (Yinsen) Zhang.
+// Use of this source code is governed by the MIT license that can be found in the LICENSE.md file.
+package org.aya.cli.repl;
+
+import com.google.gson.*;
+import kala.collection.immutable.ImmutableSeq;
+import kala.collection.mutable.MutableMap;
+import kala.control.Result;
+import org.aya.pretty.printer.ColorScheme;
+import org.aya.pretty.printer.StyleFamily;
+import org.aya.pretty.style.AyaColorScheme;
+import org.aya.util.reporter.Problem;
+import org.aya.util.reporter.Reporter;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.lang.reflect.Type;
+import java.util.Map;
+
+public record RenderOptions(
+  @NotNull ColorScheme colorScheme,
+  @NotNull StyleFamily styleFamily
+) {
+  public final static class Deserializer implements JsonDeserializer<RenderOptions>, JsonSerializer<RenderOptions> {
+    public static class ColorSchemeJson {
+      public @Nullable java.util.Map<String, String> definedColors;
+    }
+
+    public static class RenderOptionsJson {
+      public @Nullable ColorSchemeJson colorScheme;
+      public @Nullable JsonElement styleFamily;    // TODO
+    }
+
+    public @NotNull Reporter reporter;
+    public @NotNull RenderOptions fallback;
+
+    public Deserializer(@NotNull Reporter reporter, @NotNull RenderOptions fallback) {
+      this.reporter = reporter;
+      this.fallback = fallback;
+    }
+
+    public @NotNull ColorScheme colorSchemeFromJson(@NotNull ColorSchemeJson options) {
```

Comment:
Maybe use a data class directly? I don't think we'll be having custom colors

Reply:
Did you mean the builtin color schemes only?

Reply:
I'm talking about the keys. As I understood, a color scheme is a set of key-value pairs where keys are some names and values are colours (or other attributes). So, we can simply make `class ColorScheme { Style fnName; Style dataName; ... }` and use Gson to deserialize it and then use it in the pretty printer